etl/steps/data/garden/artificial_intelligence/2023-07-25/cset.py

```diff
@@ -0,0 +1,83 @@
+"""Load a meadow dataset and create a garden dataset."""
+
+from typing import cast
+
+import pandas as pd
+import us_cpi
+from owid.catalog import Dataset, Table
+from structlog import get_logger
+
+from etl.data_helpers import geo
+from etl.helpers import PathFinder, create_dataset
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+log = get_logger()
+
+
+def run(dest_dir: str) -> None:
+    #
+    # Load inputs.
+    #
+    # Load meadow dataset.
+    ds_meadow = cast(Dataset, paths.load_dependency("cset"))
+
+    # Read table from meadow dataset.
+    tb = ds_meadow["cset"]
+    tb.reset_index(inplace=True)
+    #
+    # Process data.
+    #
+    tb: Table = geo.harmonize_countries(df=tb, countries_file=paths.country_mapping_path)
+
+    # Grouping the DataFrame by 'year' to get the sum for each year
+    df_total = tb.groupby(["year", "field"]).sum(numeric_only=True)
+
+    # Resetting the index to convert 'year' from the index to a column
+    df_total.reset_index(inplace=True)
+
+    # Adding a 'Total' row in the 'country' column
+    df_total["country"] = "Total"
+    df_total.set_index(["year", "country", "field"], inplace=True)
+    merged_total = pd.concat([df_total, tb])
+    # List of columns to include for conversion to millions (investment values)
+    _investment_cols = [col for col in merged_total.columns if "investment" in col]
+    # Convert all other columns to million
+    merged_total.loc[:, _investment_cols] *= 1e6
+
+    # Import US CPI data from the API
+    df_wdi_cpi_us = us_cpi.import_US_cpi_API()
```

Comment:
Wow this is super cool, you're adjusting the investment variables to account for inflation?